towerlib/towerlib.py

```diff
@@ -1997,24 +2011,579 @@ def schedules(self):
+                playbooks = project.playbooks
+                break
+        return playbooks
+
+    def get_production_branch_playbooks(self, scm_url):
+        """Get all the playbooks of the project, which has the branch production as scm branch.
+
+        Args:
+            scm_url: given scm url
+
+        Returns:
+            list: a list with the names of all the playbooks of production branch project.
+
+        """
+        projects = list(self.projects)
+        for project in projects:
+            if project.scm_url == scm_url and project.scm_branch == 'production':
+                playbooks = project.playbooks
+                return playbooks
+
+    def get_ansible_facts_by_host_id(self, host_id):
+        """Get the ansible_facts of the given host.
+
+        Args:
+            host_id: id of the host for which the method will return the ansible_facts.
+
+        Returns:
+            dict: ansible_facts as dictionary.
+
+        """
+        host = self.get_host_by_id(host_id)
+        return host.ansible_facts()
+
+    def get_hosts_by_inventory_id(self, inventory_id):
+        """Get filtered list of hosts for a given inventory id.
+
+        Args:
+             inventory_id: the given inventory id.
+
+        Returns:
+            list: the list of filtered host as EntityManger object.
+
+        """
+        hosts = [item for item in self.hosts if item.inventory.id == inventory_id]
+        return hosts
+
+    def get_jobs_by_name(self, given_job_name):
+        """Get filtered list of jobs for a given name.
+
+        Args:
+            given_job_name: the given job name.
+
+        Returns:
+             list: the filtered list of jobs.
+
+        """
+        job_list = [item for item in self.jobs if item.name == given_job_name]
+        return job_list
+
+    def get_project_updates(self):
+        """Get all the project updates for the ansible tower.
+
+        Returns:
+            list: list of all the project updates.
+
+        """
+        project_updates_url = '{api}/project_updates/'.format(api=self.api)
+        response = self.session.get(project_updates_url)
+        if not response.ok:
+            self._logger.error("Error getting the project updates. response was: {})".format(response.text))
+            return None
+        else:
+            return response.json().get('results', [])
+
+    def get_project_updates_by_project_name(self, given_project_name):
+        """Get project update with the given project name.
+
+        Args:
+            given_project_name: the name of the project.
+
+        Returns:
+            dict: the project with all the information.
+
+        """
+        project_update_list = self.get_project_updates()
+        project_updates = [item for item in project_update_list if item['name'] == given_project_name]
+        return project_updates
+
+    def get_project_updates_by_project_id(self, given_project_id):
```

Comment:
This should be implemented as a filter on the project_updates object after it has been implemented.

Reply:
Implemented.